docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,147 @@
+# Envoy Gateway Extensions Design
+
+As outlined in the [official goals][] for the Envoy Gateway project, one of the main goals is to "provide a common foundation for vendors to build value-added products 
+without having to re-engineer fundamental interactions". Development of the Envoy Gateway project has been focused on developing the core features for the project and 
+K8s Gateway API conformance. This system focuses on the “common foundation for vendors” component by introducing a way for vendors to extend Envoy Gateway.
+
+## Goals
+
+- Provide a foundation for extending the Envoy Gateway control plane
+- Allow users to attach [extensionRef HTTPRoute filters][] to resources introduced and handled by Envoy Gateway extensions.
+- Extension developers should **NOT** have to maintain a custom fork of Envoy Gateway
+- Extension developers should **NOT** require expert knowledge of all of Envoy Gateway's 
+internal components to meaningfully extend the control plane.
+- Provide a system for extending Envoy Gateway which allows extension projects to ship updates independent of Envoy Gateway's release schedule
+- Extensions can hook into both the Envoy configuration pipeline and the infra manager pipeline
+- Setup a foundation for the initial iteration of Extending Envoy Gateway
+
+## Non-Goals
```

Comment:
Another non-goal to potentially add is not fully reconciling the multiple extensions case